### PR TITLE
feat(TGI): add release docker image build and push to registry workflow

### PR DESCRIPTION
**Pull request by tengomucho**
# What does this PR do?

Prepares CI to upload TGI image on TPUs.

### CODE REVIEW

File: .github/workflows/tpu-tgi-release.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          uses: docker/build-push-action@v6
+          with:
+            context: .
+            file: text-generation-inference/docker/Dockerfile
+            push: true
+            tags: ${{ steps.meta.outputs.tags }}
+            labels: ${{ steps.meta.outputs.labels }}
+            build-args: |
+              VERSION=${{ steps.version.outputs.version }}
+              TGI_VERSION=5bc3d65dd32ba1f979540caeccbf3dd8798dd9df
```
Comment (oOraph) [inline .github/workflows/tpu-tgi-release.yml:76]: Is TGI_VERSION supposed to be frozen here ?

File: .github/workflows/tpu-tgi-release.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          uses: docker/build-push-action@v6
+          with:
+            context: .
+            file: text-generation-inference/docker/Dockerfile
+            push: true
+            tags: ${{ steps.meta-ie.outputs.tags }}
+            labels: ${{ steps.meta-ie.outputs.labels }}
+            build-args: |
+              VERSION=${{ steps.version.outputs.version }}
+              TGI_VERSION=5bc3d65dd32ba1f979540caeccbf3dd8798dd9df
```
Comment (oOraph) [inline .github/workflows/tpu-tgi-release.yml:97]: same question as above

**Review by oOraph — APPROVED** [review]

File: .github/workflows/tpu-tgi-release.yml (hunk 1 above)
Comment (tengomucho) [reply .github/workflows/tpu-tgi-release.yml:76]: short answer: yes, for now. Long answer: in the future, TPU TGI development should move into TGI repo, so image building and version will be different. But I guess this should be fine for now.

File: .github/workflows/tpu-tgi-release.yml (hunk 2 above)
Comment (tengomucho) [reply .github/workflows/tpu-tgi-release.yml:97]: same answer

**Review by mfuntowicz — APPROVED** [review]

**Pull request merged**